src/display/canvas.js

```diff
@@ -465,14 +465,14 @@ function compileType3Glyph(imgData) {
     0, 2, 4, 0, 1, 0, 5, 4, 8, 10, 0, 8, 0, 2, 1, 0,
   ]);
 
-  const width1 = width + 1,
-    points = new Uint8Array(width1 * (height + 1));
+  const width1 = width + 1;
+  let points = new Uint8Array(width1 * (height + 1));
   let i, j, j0;
 
   // decodes bit-packed mask data
-  const lineSize = (width + 7) & ~7,
-    data = new Uint8Array(lineSize * height);
-  let pos = 0;
+  const lineSize = (width + 7) & ~7;
+  let data = new Uint8Array(lineSize * height),
```

Comment:
Maybe I'm overlooking something, but `data` is just used to "uncompressed" `imgData.data` and then we read its values to build the path.
Would it make sense to remove `data` use and then rewrite the algorithm to directly use `imgData.data` ?
Likely it'd increase code complexity and maybe it could be a perf penalty... Anyway, it's just a suggestion.